### PR TITLE
[internal/datadog] Fix Datarace in metrics client

### PR DESCRIPTION
**Pull request by dineshg13**
**Description:** <Describe what has changed.>

Fixes the data race.
```
fatal error: concurrent map read and map write

goroutine 72665 [running]:
github.com/open-telemetry/opentelemetry-collector-contrib/internal/datadog.(*metricsClient).Gauge.func1({0x7b895a0?, 0xc07c930790?}, {0x940dac8, 0xc1996ce180})
	github.com/open-telemetry/opentelemetry-collector-contrib/internal/datadog@v0.96.0/metrics_client.go:49 +0x85
go.opentelemetry.io/otel/sdk/metric.(*meter).float64ObservableInstrument.func1.1({0x9443f08, 0xef10b00})
	go.opentelemetry.io/otel/sdk/metric@v1.24.0/meter.go:286 +0x55
go.opentelemetry.io/otel/sdk/metric.(*pipeline).produce(0xc002852bd0, {0x9443f08, 0xef10b00}, 0xc108dd46c0)
	go.opentelemetry.io/otel/sdk/metric@v1.24.0/pipeline.go:122 +0x168
go.opentelemetry.io/otel/sdk/metric.(*ManualReader).Collect(0xc0030d2aa0, {0x9443f08, 0xef10b00}, 0xc108dd46c0)
	go.opentelemetry.io/otel/sdk/metric@v1.24.0/manual_reader.go:123 +0xe2
go.opentelemetry.io/otel/exporters/prometheus.(*collector).Collect(0xc0030e4aa0, 0xc10a0083c0)
	go.opentelemetry.io/otel/exporters/prometheus@v0.46.0/exporter.go:158 +0x72
github.com/prometheus/client_golang/prometheus.(*Registry).Gather.func1()
	github.com/prometheus/client_golang@v1.19.0/prometheus/registry.go:457 +0xe5
created by github.com/prometheus/client_golang/prometheus.(*Registry).Gather in goroutine 5
```

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

File: internal/datadog/metrics_client_test.go [hunk 1]
```diff
@@ -72,6 +74,44 @@ func TestGaugeMultiple(t *testing.T) {
...
+	reader, metricClient, _ := setupMetricClient()
+	var wg sync.WaitGroup
+	wg.Add(2)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	go func() {
+		defer wg.Done()
+		cnt := 0
```
Comment (liustanley) [inline internal/datadog/metrics_client_test.go:87]: What is the purpose of `cnt`? Is it supposed to be incremented in the for loop or remain as a constant?

**Comment by mx-psi** [comment]
You will need to rebase this one after #31972 is merged

**Review by songy23 — COMMENTED** [review]
Does this need a changelog? Seems users can run into this bug

**Comment by dineshg13** [comment]
@songy23 I would prefer not to add changelog for a data race bug.

**Review by liustanley — APPROVED** [review]

**Comment by mx-psi** [comment]
I agree with Yang that this would need a changelog, I mistakenly assumed this did not have user-facing implications

**Review by mx-psi — APPROVED** [review]

**Review by songy23 — APPROVED** [review]

File: .chloggen/dinesh.gurumurthy_fix-data-race-metrics-client.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: [datadog/exporter, datadog/connector]
```
Comment (songy23) [inline .chloggen/dinesh.gurumurthy_fix-data-race-metrics-client.yaml:6]:
```suggestion
component: datadog/connector
```

Comment (songy23) [reply .chloggen/dinesh.gurumurthy_fix-data-race-metrics-client.yaml:6]: Seems you can only have one component per changelog entry

**Pull request merged**